go-controller/pkg/ovn/gateway_init.go

```diff
@@ -170,20 +190,51 @@ func gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet, hostSubnets []*n
 					"gateway router %s, stdout: %q, stderr: %q, error: %v",
 					gatewayRouter, stdout, stderr, err)
 			}
-		}
-		// Also add north-south load-balancers to local switches for pod -> nodePort traffic
-		stdout, stderr, err = util.RunOVNNbctl("get", "logical_switch", nodeName, "load_balancer")
-		if err != nil {
-			return fmt.Errorf("failed to get load-balancers on the node switch %s, stdout: %q, "+
-				"stderr: %q, error: %v", nodeName, stdout, stderr, err)
-		}
-		for _, proto := range enabledProtos {
-			if !strings.Contains(stdout, protoLBMap[proto]) {
-				stdout, stderr, err = util.RunOVNNbctl("ls-lb-add", nodeName, protoLBMap[proto])
-				if err != nil {
-					return fmt.Errorf("failed to add north-south load-balancer %s to the "+
-						"node switch %s, stdout: %q, stderr: %q, error: %v",
-						protoLBMap[proto], nodeName, stdout, stderr, err)
+
+			// Ensure north-south load-balancers are not on local switches for pod -> nodePort traffic
+			// For upgrade path REMOVEME later
+			stdout, stderr, err = util.RunOVNNbctl("get", "logical_switch", nodeName, "load_balancer")
+			if err != nil {
+				return fmt.Errorf("failed to get load-balancers on the node switch %s, stdout: %q, "+
+					"stderr: %q, error: %v", nodeName, stdout, stderr, err)
+			}
```

Comment:
can we use https://github.com/ovn-org/ovn-kubernetes/blob/a538127829e5c18110cbf01470c2cfbaaa09c155/go-controller/pkg/ovn/loadbalancer/loadbalancer.go#L79-L80 instead.

This will simplify the migration to bindings later, since it will only require to replace the nbctl call in the ovn/loadbalancer package, and will make the code more consistent

Reply:
if it returns  map for protos you can avoid the strings matching later

Reply:
I agree it would be an improvement, but I would like to do that in a follow up (and perhaps even migrate some of this to go-ovn bindings)